src/modal/__integ__/modal.test.ts

```diff
@@ -114,3 +115,26 @@ test(
     await expect(page.isExisting('#async-modal-root')).resolves.toBe(false);
   })
 );
+
+test(
+  'verifies modal performance metrics are emitted after all components are loaded',
+  useBrowser(async browser => {
+    const page = new BasePageObject(browser);
+    await browser.url('#/light/modal/with-component-load');
+    const getModalPerformanceMetrics = () =>
+      browser.execute(() => ((window as any).modalPerformanceMetrics ?? []) as ModalPerformanceDataProps[]);
+    await page.click('[data-testid="modal-trigger"]');
+    let metrics = await getModalPerformanceMetrics();
+
+    //verify metrics are not emitted until all the components are loaded
+    expect(metrics.length).toBe(0);
+
+    await delay(2000);
```

Comment:
can we have a more specific wait? 
what components are you waiting for to be ready? you can use `await page.waitForVisible` or `await waitFor` instead of specifying a time